pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -86,19 +104,75 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 		return err
 	}
 
-	mapper, typer := f.Object()
-	// TODO: use resource.Builder instead
-	obj, err := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
-		NamespaceParam(cmdNamespace).RequireNamespace().
-		FilenameParam(filename).
-		Do().
-		Object()
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+
+	// fetch rc
+	oldRc, err := client.ReplicationControllers(cmdNamespace).Get(oldName)
 	if err != nil {
 		return err
 	}
-	newRc, ok := obj.(*api.ReplicationController)
-	if !ok {
-		return cmdutil.UsageError(cmd, "%s does not specify a valid ReplicationController", filename)
+
+	mapper, typer := f.Object()
+	var newRc *api.ReplicationController
+
+	if len(filename) != 0 {
+		obj, err := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+			NamespaceParam(cmdNamespace).RequireNamespace().
+			FilenameParam(filename).
+			Do().
+			Object()
+		if err != nil {
+			return err
+		}
+		var ok bool
+		newRc, ok = obj.(*api.ReplicationController)
+		if !ok {
+			return cmdutil.UsageError(cmd, "%s does not specify a valid ReplicationController", filename)
+		}
+	}
+	if len(image) != 0 {
+		if len(args) < 2 {
+			return cmdutil.UsageError(cmd, "You must specify a name for your new controller")
+		}
+		newName := args[1]
+		var err error
+		// load the old RC into the "new" RC
+		if newRc, err = client.ReplicationControllers(cmdNamespace).Get(oldName); err != nil {
+			return err
+		}
+
+		if len(newRc.Spec.Template.Spec.Containers) > 1 {
+			// TODO: support multi-container image update.
+			return cmdutil.UsageError(cmd, "Image update is not supported for multi-container pods")
```

Comment:
I'm not sure `UsageError` is appropriate here as opposed to just a regular error.  The command had the right arguments.